Update ForwardingRule and GlobalForwardingRule to add pscConnectionId…

muresan opened the pull request:
… and pscConnectionStatus attributes

<!-- Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to -->
Fixes https://github.com/hashicorp/terraform-provider-google/issues/10588



<!--
Replace each [ ] with [X] to check it. Switch to the preview view to make it easier to click on links.
These steps will speed up the review process, and we appreciate you spending time on them before sending
your code to be reviewed.
-->
If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

<!-- AUTOCHANGELOG for Downstream PRs.

Please select one of the following "release-note:" headings:
    - release-note:enhancement
    - release-note:bug
    - release-note:note
    - release-note:new-resource
    - release-note:new-datasource
    - release-note:deprecation
    - release-note:breaking-change
    - release-note:none
    
Unless you choose release-note:none, please add a release note.

See .ci/RELEASE_NOTES_GUIDE.md for writing good release notes.

You can add more release note blocks if you want more than one CHANGELOG
entry for this PR.
-->
**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: add psc_connection_id and psc_connection_status to google_compute_forwarding_rule and google_compute_global_forwarding_rule
```

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6106-old..auto-pr-6106) ( 2 files changed, 18 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6106-old..auto-pr-6106) ( 2 files changed, 18 insertions(+))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-6106-old..auto-pr-6106) ( 4 files changed, 23 insertions(+), 3 deletions(-))

ScottSuarez commented:
could you follow the steps to sign the CLA?

muresan commented on mmv1/products/compute/api.yaml:
```diff
@@ -4114,6 +4114,23 @@ objects:
+          - :PENDING
+          - :REJECTED
+          - :STATUS_UNSPECIFIED
+          - :UNDEFINED_PSC_CONNECTION_STATUS
```
Does not look like it is settable, documentation is incomplete, the console has no options to set this field. Changed to String and output: true.

muresan commented:
> could you follow the steps to sign the CLA?

CLA shows signed on my end (and the bot also shows OK).

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6106-old..auto-pr-6106) ( 2 files changed, 14 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6106-old..auto-pr-6106) ( 2 files changed, 14 insertions(+))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-6106-old..auto-pr-6106) ( 2 files changed, 3 insertions(+), 3 deletions(-))

ScottSuarez commented on mmv1/products/compute/api.yaml:
```diff
@@ -4114,6 +4114,17 @@ objects:
+      - !ruby/object:Api::Type::String
+        name: 'pscConnectionId'
+        description: |
+          [Output Only] The PSC connection id of the PSC Forwarding Rule in uint64 format.
```
```suggestion
          The PSC connection id of the PSC Forwarding Rule in uint64 format.
```

Remove [Output Only], documentation automation handles this

ScottSuarez commented on mmv1/products/compute/api.yaml:
```diff
@@ -4491,6 +4502,17 @@ objects:
+      - !ruby/object:Api::Type::String
+        name: 'pscConnectionId'
+        description: |
+          [Output Only] The PSC connection id of the PSC Forwarding Rule in uint64 format.
```
```suggestion
          The PSC connection id of the PSC Forwarding Rule in uint64 format.
```

muresan replied:
Updated

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6106-old..auto-pr-6106) ( 2 files changed, 14 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6106-old..auto-pr-6106) ( 2 files changed, 14 insertions(+))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-6106-old..auto-pr-6106) ( 2 files changed, 3 insertions(+), 3 deletions(-))

ScottSuarez commented:
/gcbrun

modular-magician commented:
Hi!  I'm the modular magician.  Your PR generated some diffs in downstreams - here they are.
# Diff report:


Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-6106-old..auto-pr-6106) ( 2 files changed, 14 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-6106-old..auto-pr-6106) ( 2 files changed, 14 insertions(+))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-6106-old..auto-pr-6106) ( 2 files changed, 3 insertions(+), 3 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `2038` 
Passed tests `1809` 
Skipped tests: `226` 
Failed tests: `3` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary> TestAccContainerCluster_withConfidentialNodes|TestAccContainerCluster_withAddons|TestAccSqlUser_mysqlDisabled </details>

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccSqlUser_mysqlDisabled`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6106/artifacts/60ca9f79-27b5-46f2-872c-ed11e305b06c/recording/TestAccSqlUser_mysqlDisabled.log)] 

Tests failed during RECORDING mode:
 `TestAccContainerCluster_withAddons`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6106/artifacts/60ca9f79-27b5-46f2-872c-ed11e305b06c/recording/TestAccContainerCluster_withAddons.log)]
`TestAccContainerCluster_withConfidentialNodes`[[view](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6106/artifacts/60ca9f79-27b5-46f2-872c-ed11e305b06c/recording/TestAccContainerCluster_withConfidentialNodes.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-6106/artifacts/60ca9f79-27b5-46f2-872c-ed11e305b06c/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-6106/artifacts/60ca9f79-27b5-46f2-872c-ed11e305b06c/recording) for each test

muresan commented:
> So unfortunately it looks like these resources are generated by the DCL which is a library that our team doesn't directly own. Thus we are unable to surface these fields :/
> 
> If you look at the generated providers it seems only the documentation is getting updated.

I got here from https://github.com/hashicorp/terraform-provider-google/pull/11767 which lead to https://github.com/GoogleCloudPlatform/declarative-resource-client-library/pull/11 which lead me to create this PR. As you can see from this comment @megan07  https://github.com/hashicorp/terraform-provider-google/pull/11767#issuecomment-1137459731 magic-modules was mentioned as the source for DCL (my assumption here is that by DCL you mean GoogleCloudPlatform/declarative-resource-client-library, if not please correct me)

What are the options ? since the https://github.com/GoogleCloudPlatform/declarative-resource-client-library/blob/main/docs/contributing.md for declarative-resource-client-library clearly offers to options.

Chicken and Egg problem ?

muresan commented:
> If you look at the generated providers it seems only the documentation is getting updated.

I ran locally the command to generate the terraform code:
```
make terraform VERSION=ga OUTPUT_PATH="$GOPATH/src/github.com/hashicorp/terraform-provider-google"
``
and it _did_ generate the code but at the very end there was a warning and then it reverted back all the code. If you execute `git status` in the terraform provider source code before it completes but after it generates Compute you will see the code there. That's why I did not check above the Generate Terraform option, assumed my local ruby (or something) had issues.

```
12:32 $ git status|grep  compute_ | grep -v website
	modified:   google/resource_compute_forwarding_rule.go
	modified:   google/resource_compute_forwarding_rule_sweeper_test.go
	modified:   google/resource_compute_global_forwarding_rule.go
	modified:   google/resource_compute_global_forwarding_rule_sweeper_test.go
```
Here are the generated files: https://gist.github.com/muresan/7c1aca765407eb9fb6611e1c02c8aaab
They "disappear" at the end, the output form the run is also in the gist

Not sure how to debug this further

jamesagarside commented:
Thank you @muresan for creating this pull request. I am very keen for this functionality as its currently a blocker on a project of mine. 
Is there anything I can help with?

ScottSuarez commented:
We're working to add this to the DCL. It should make it's way downstream within the next few weeks.

muresan commented:
> We're working to add this to the DCL. It should make it's way downstream within the next few weeks.

Looks like Private Service Connect structures were recently added to the DCL: https://github.com/GoogleCloudPlatform/declarative-resource-client-library/commit/785f52764dd60f0beb449f590acbb3a4c7c16639 , v1.11.0 has the required resources, what would be the next steps ?

Thanks

ScottSuarez commented:
The next step would be updating the DCL package by running `make upgrade dcl`

Looks like the package has already been updated 
https://github.com/GoogleCloudPlatform/magic-modules/pull/6124

muresan commented:
> The next step would be updating the DCL package by running `make upgrade dcl`
> 
> Looks like the package has already been updated #6124

Closing this PR as it's useless now. Curious why this change was not be merged since it's identical in functinality to what was merged.

razine-bensari commented:
Is it possible to reopen this ?
I think this feature is quite useful.
Why was it closed ?
